src/node_crypto.cc

```diff
@@ -2590,10 +2590,12 @@ void CipherBase::Init(const char* cipher_type,
                                1,
                                key,
                                iv);
+  CHECK_NE(key_len, 0);
 
   ctx_.reset(EVP_CIPHER_CTX_new());
   const bool encrypt = (kind_ == kCipher);
-  EVP_CipherInit_ex(ctx_.get(), cipher, nullptr, nullptr, nullptr, encrypt);
+  CHECK(EVP_CipherInit_ex(ctx_.get(), cipher, nullptr,
```

Comment:
`EVP_CipherInit_ex()` can fail for legitimate reasons (e.g. an engine that fails to load) so a `CHECK` is not appropriate.

Reply:
I assume the old behavior (= no error checking) isn't appropriate either?

Reply:
Indeed, errors should be reported.

Reply:
That would make this semver-major I guess, so I'll probably just leave the `CHECK`s out for now.

Reply:
Mwah, I don't think better error checking constitutes semver-major.  The kind of bugs it'd flush out are programmer bugs and environmental issues.